### DISCUSSION TITLE
Add package to a testcontainer on its setup

### DISCUSSION THREAD

**Discussion by vglinca** [testcontainers/testcontainers-dotnet#1075] [Q&A] [answered]
Hi,
I need to install a full-text-search package for the MsSqlServer container, but I'm stuck on doing that.
To install the package, this command is needed: `apt-get install -y mssql-server-fts`. And I try to execute it during the container creation:
`MsSqlContainer _dbContainer = new MsSqlBuilder().WithCommand("apt-get install -y mssql-server-fts").Build();`
but it doesn't work. What have I missed?

**Comment by HofmeisterAn** [comment, marked as answer]
This is not how Testcontainers and its modules work. The command API is used to specify the command that the container starts with. Even if it installs the packages, it would exit afterward since the database is not started. Build and use a [custom](https://stackoverflow.com/a/60525839/690017) image instead (`Dockerfile`).

    _ = new MsSqlBuilder().WithImage("mssql:fts.2019").Build();

**Reply by vglinca** [reply]
It's a good idea, thank you. But it still doesn't work, unfortunately...
I've built the custom image. It works when I run it using the `docker run` command. And I'm able to connect to the server too. 
Though when I build a test container, it just waits forever for an awaited method `StartAsync()` to complete. In docker desktop I can see a newly created container, and there are no errors in the logs, moreover, it logs out that the server is available at specified host. But I can't connect to it

**Reply by HofmeisterAn** [reply]
> it just waits forever for an awaited method `StartAsync()` to complete.

If that is the case, then the default wait strategy is either incompatible or your newly built image is broken ([wait strategy](https://github.com/testcontainers/testcontainers-dotnet/blob/develop/src/Testcontainers.MsSql/MsSqlBuilder.cs#L132-L144) never completes).
